src/main/java/games/strategy/util/Match.java

```diff
@@ -172,4 +185,79 @@ public boolean match(final T value) {
       }
     };
   }
+
+  /**
+   * Creates a new match whose condition is satisfied if the test object matches any of the specified conditions.
+   *
+   * @param matches An array of matches; must not be {@code null}.
+   *
+   * @return A new match; never {@code null}.
+   */
+  @SafeVarargs
+  public static <T> Match<T> any(final Match<T>... matches) {
+    checkNotNull(matches);
+
+    return any(Arrays.asList(matches));
```

Comment:
> I'm aiming to replace those classes with the stream API

That's effectively what this change (for `CompositeMatchOr`) and a forthcoming change (for `CompositeMatchAnd`) are doing.  A `Match` is really just a `Predicate`, and a composite `Match` is just a collection of `Predicates` that uses `Stream#anyMatch()` or `Stream#allMatch()` to evaluate the condition.  Actually, by the end of this series of PRs, there should be no need for `Match` subclasses at all (anonymous or otherwise).

Were you thinking of something different when you said, "replace those classes with the stream API?"